src/nirfsg/system_tests/test_system_nirfsg.py

```diff
@@ -0,0 +1,559 @@
+import array
+import hightime
+import nirfsg
+import numpy as np
+import os
+import pathlib
+import pytest
+import sys
+import time
+
+
+sys.path.insert(0, str(pathlib.Path(__file__).parent.parent.parent / 'shared'))
+import system_test_utilities  # noqa: E402
+
+# Set up global information we need
+test_files_base_dir = os.path.join(os.path.dirname(__file__))
+use_simulated_session = True
+real_hw_resource_name = '5841'
+
+
+def get_test_file_path(file_name):
+    return os.path.join(test_files_base_dir, file_name)
+
+
+sys.path.insert(0, str(pathlib.Path(__file__).parent.parent.parent / 'generated/nirfsg'))
+
+
+class SystemTests:
+    @pytest.fixture(scope='function')
+    def test_rfsg_device_session(self, session_creation_kwargs):
+        if use_simulated_session:
+            with nirfsg.Session("5841sim", options="Simulate=1, DriverSetup=Model:5841", **session_creation_kwargs) as sim_5841_session:
+                yield sim_5841_session
+        else:
+            with nirfsg.Session(real_hw_resource_name, options="", **session_creation_kwargs) as real_rfsg_device_session:
```

Comment:
```suggestion
            with nirfsg.Session(real_hw_resource_name, **session_creation_kwargs) as real_rfsg_device_session:
```

Nitpick: options is an optional argument. I would omit it, since you're not passing any.